x-pack/plugins/ingest_manager/common/services/agent_status.ts

```diff
@@ -19,6 +19,9 @@ export function getAgentStatus(agent: Agent, now: number = Date.now()): AgentSta
   if (!agent.last_checkin) {
     return 'enrolling';
   }
+  if (agent.upgrade_started_at && !agent.upgraded_at) {
```

Comment:
this won't work once the user upgrades a second time, unless we reset the values.  However we will be sending "upgrade" as an agent state event in a future PR, so that will probably be used instead.